### DISCUSSION TITLE
Is there a way to update or delete a message in a stream?

### DISCUSSION THREAD

**Discussion by vamshiaruru** [nats-io/nats-server#4883] [Q&A] [answered]
Hi everyone, in nats jetstream, is there a way to update or delete a message? Some youtube video I saw implied the possibility, but I couldn't find any documentation for this. Thanks!

**Comment by derekcollison** [comment, marked as answer]
You can not change a message once committed to a stream, but you can delete, and even opt to securely delete a message by sequence.

You can also purge message by subject, including wildcards.

**Reply by vamshiaruru** [reply]
Where can I find the documentation to dleete a message by sequence?

**Reply by derekcollison** [reply]
Docs or READMEs for the client language of your choice. Or if using the NATS cli, `nats stream rmm` which will securely delete a message.

**Reply by vamshiaruru** [reply]
I have tried going through go's client documentation to figure out what function to use, but couldn't find the function. But I went through the nats cli code to see what the cli is calling, and that lead me to https://pkg.go.dev/github.com/nats-io/jsm.go . So for anyone who is interested, this is the place you have to check.

**Reply by ripienaar** [reply]
https://github.com/nats-io/nats.go/blob/b95565dc9a851ea318f1c0bc83b312c65a29a6a7/jetstream/stream.go#L47

**Reply by vamshiaruru** [reply]
Thanks! Not sure how I couldn't findd it in the docs, much appreciated.
